hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/InitMetaProcedure.java

```diff
@@ -119,6 +122,19 @@ protected Flow executeFromState(MasterProcedureEnv env, InitMetaState state)
           // namespaces when starting.
           insertNamespaceToMeta(env.getMasterServices().getConnection(), DEFAULT_NAMESPACE);
           insertNamespaceToMeta(env.getMasterServices().getConnection(), SYSTEM_NAMESPACE);
+          // create the additional namespaces based on the configuration value
+          String[] customNamespaces =
+            env.getMasterConfiguration().getStrings(ADDITIONAL_NAMESPACES);
+          if (null != customNamespaces && customNamespaces.length > 0) {
+            Arrays.stream(customNamespaces).forEach(ns -> {
```

Comment:
Why not just follow the regular way to create these namespaces?

We have to insert system namespace here because meta table is in this namespace, so there is a cyclic dependency...

For other namespaces, I suggest to create them with your own operational scripts or some management systems, from outside the cluster.

Reply:
This improvement provides a config option for operators to create the additional namespaces during the HBase startup. This would eliminate the need to wrap createTable calls in checkAndCreateNamespace or provide separate cluster bootstrap functionality to guarantee that the namespace is created.

Reply:
But the code can only take effect when initializing the cluster, i.e, on the first startup of this HBase cluster... If later you changed the configuration and restart the hbase cluster, it will not take effect.

I think this will confuse our users...